pandas/tests/scalar/test_na_scalar.py

```diff
@@ -267,3 +269,29 @@ def test_integer_hash_collision_set():
     assert len(result) == 2
     assert NA in result
     assert hash(NA) in result
+
+
+def test_pickle_roundtrip():
+    # https://github.com/pandas-dev/pandas/issues/31847
+    result = pickle.loads(pickle.dumps(pd.NA))
+    assert result is pd.NA
+
+
+def test_pickle_roundtrip_pandas():
+    with tm.ensure_clean("data.pkl") as f:
+        with open(f):
+            pd.to_pickle(pd.NA, f)
+        with open(f):
+            result = pd.read_pickle(f)
+    assert result is pd.NA
+
+
+def test_pickle_roundtrip_series():
+    s = pd.Series(pd.array([1, 2, pd.NA]))
```

Comment:
Can you also add a test with object array + one with string array? As the integer array doesn't actually store pd.NA, while those others do

Reply:
(to also cover https://github.com/pandas-dev/pandas/issues/32095, and https://github.com/pandas-dev/pandas/issues/31847#issuecomment-588500110)